consensus/core/src/authority_node.rs

```diff
@@ -540,7 +540,7 @@ mod tests {
 
     // TODO: build AuthorityFixture.
     #[rstest]
-    #[tokio::test(flavor = "current_thread", start_paused = true)]
```

Comment:
I tried this yesterday as a potential mitigation but it increased the test runtime +60 seconds. I was thinking maybe the solution was to use a MockClock instance for tests, but would require some refactoring so you could inject it or maybe on initialization check some test flag?

Reply:
Looking deeper the issue comes from the fact that the `std::thread::sleep(Duration::from_millis(wait_ms));` on Core during recovery is not advancing the clock leading to a panic when the timestamp checks are done. Now it looks like that that only the `Tonic` test fails and never the `Anemo`. Taking a closer look the `Anemo` test seems to run in real time - ignoring the `start_paused = true` direction (so that on its own it takes even today 60 seconds to run). I've investigated the reason for that but didn't find something too obvious. I am timeboxing the investigation so can merge those changes for now. Will follow up later. For now I've reduced the sleeps so both tests run within ~35 sec.